File: test/client-c#/Exemple.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Influence
{
    class Exemple
    {
        static void Main(string[] args)
        {
			// Initialisation du random
            Random r = new Random();
			
			// On récupère l'instance du client (cela le crée en fait)
            InfluenceClient client = InfluenceClient.GetInstance();

			// On se connecte au serveur : le premier paramètre est son adresse IP, le deuxième est le nom de votre équipe
			// Cet appel affiche votre numéro sur le plateau de jeu
            client.Connect("127.0.0.1", "test1");

			// Tant que la partie est en cours
            while (client.GetStatus() == InfluenceClient.Status.ONGOING)
            {
				// On attend notre tour, ce qui permet de récupérer le nouveau plateau de jeu
                InfluenceField field = client.NextRound();

				 // Après l'appel à nextRound, il se peut que le statut ait été modifié
                if (client.GetStatus() != InfluenceClient.Status.ONGOING)
                {
                    break;
                }

                List<InfluenceCell> myCells;
                Console.WriteLine(DateTime.Now + " - Attacking");
				
				// On va essayer de lancer 10 attaques
                for (int i = 0; i < 10; i++)
                {
					// On récupère toutes les cellules qui nous appartiennent
                    myCells = client.GetMyCells();
					
					// On en choisit une aléatoirement parmi celles-ci
                    InfluenceCell c = myCells.ElementAt(r.Next(myCells.Count));
					
					// Si la cellule choisie possède la puissance nécessaire pour attaquer
                    if (c.GetUnitsCount() >= 2)
                    {
						// On définit une cellule cible dans son entourage
                        int dx = c.GetX() + r.Next(3) - 1;
                        int dy = c.GetY() + r.Next(3) - 1;
						
						// Si la cible est une position valide (ses coordonnées sont bien sur le plateau)
                        if (dx >= 0 && dx < field.GetWidth() && dy >= 0 && dy < field.GetHeight())
                        {
                            InfluenceCell cellToAttack = field.GetCell(dx, dy);
							// Si la cellule cible appartient à un adversaire
                            if (cellToAttack != null && cellToAttack.GetOwner() != client.GetNumber())
                            {
								// On attaque la cible et on récupère le plateau de jeu après l'attaque
                                field = client.Attack(c.GetX(), c.GetY(), cellToAttack.GetX(), cellToAttack.GetY());
                            }
                        }
                    }
                }

				// On prévient le serveur qu'on a fini les attaques, il nous renvoie le nombre d'unités à placer
                int unitsToAdd = client.EndAttacks();

				// Phase de placement des unités
				
				// On récupère les cellules qui nous appartiennent
                myCells = client.GetMyCells();
				// Tant que l'on peut placer des unités
                for (int i = 0; i < unitsToAdd; i++)
                {
					// On choisit une cellule aléatoirement parmi celles qui nous appartiennent
                    InfluenceCell c = myCells.ElementAt(r.Next(myCells.Count));
					// On ajoute une unité à la cellule choisie
                    client.AddUnits(c, 1);
                }
				
				// On indique au serveur que l'on a fini notre tour
                client.EndAddingUnits();
            }
            
			// On est sorti de la boucle donc la partie est terminée
            switch (client.GetStatus())
            {
                case InfluenceClient.Status.VICTORY:
                    Console.WriteLine("YOU WON!");
                    break;

                case InfluenceClient.Status.DEFEAT:
                    Console.WriteLine("YOU LOST!");
                    break;

                case InfluenceClient.Status.CONNECTION_LOST:
                    Console.WriteLine("YOU LOST BECAUSE OF YOUR CONNECTION");
                    break;

                default:
                    Console.WriteLine("NOT REACHABLE");
                    break;
            }
			
			// Pour éviter que la console se ferme direct
            Console.ReadKey();
        }
    }
}

File: test/client-go/exemple.go
package main

import (
	"log"
	"math/rand"
	"strconv"
	"time"

	"git.ares-ensiie.eu/hackathon/hackathon-go-client/client"
)

func ce(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	// Initialisation du rng
	rng := rand.New(rand.NewSource(time.Now().UnixNano()))

	// On créé un nouveau client
	c := client.NewClient()

	// On se connecte: Le premier champ est l'IP du serveur, le second est le nom
	// de votre équipe
	ce(c.Connect("127.0.0.1:1337", "Exemple: Go"))

	// Sur le plateau, votre équipe est représentée par un identifiant numérique
	// récupéré après la phase de connection.
	log.Println("Connected ! Name: " + c.Name + ", ID: " + strconv.Itoa(c.ID))

	// Tant que le jeu est en cours
	for c.Status() == client.ONGOING {
		log.Println("Waiting for our turn")
		ce(c.NextTurn())

		// Il se peut que le status du jeu change après l'appel à la
		// fonction NextTurn.
		if c.Status() != client.ONGOING {
			break
		}

		log.Println("Attacking")

		// On va essayer de lancer 10 attaques
		for i := 0; i < 10; i++ {
			// On choisi une cellule parmis celles qui nous appartiennent
			mycell := c.MyCells()[rng.Int()%len(c.MyCells())]

			// Si cette céllule à la puissance necessaire pour attaquer
			if mycell.Power >= 2 {
				// On définit une cible dans son entourage
				dx := mycell.X + rng.Int()%3 - 1
				dy := mycell.Y + rng.Int()%3 - 1

				// Si la cible est une position valide
				if dx >= 0 && dx < c.GetField().SizeX && dy >= 0 && dy < c.GetField().SizeY {
					dest := c.Get(dx, dy)
					// Si la cible ne nous appartient pas
					if dest != nil && dest.Owner != c.ID {
						// On attaque la cible
						ce(c.Attack(mycell.X, mycell.Y, dx, dy))
					}
				}
			}
		}
		// Fin du tour d'attaque
		log.Println("End attacks")
		// On previens le serveur que l'on a finit le tour d'attaque
		_, err := c.EndAttacks()
		ce(err)

		// Début de la phase de placement des unitées
		log.Println("Adding units")
		// Tant que l'on peut placer des unitées
		for c.RemainingUnits() > 0 {

			// On choisi une céllule par hasard dans notre liste de céllules
			mycell := c.MyCells()[rng.Int()%len(c.MyCells())]
			// Et on lui ajoute une unité
			c.AddUnit(mycell)
		}
		// On indique au serveur que l'on a finit notre tour
		c.EndAddingUnits()
	}

	// On est sorti de la boucle donc la partie est terminée.

	switch c.Status() {
	case client.DEFEAT:
		log.Println("WE LOST :(")
		break
	case client.VICTORY:
		log.Println("WE WON !")
		break
	case client.CONNECTION_LOST:
		log.Println("CONNECTION LOST : WE PROBABLY LOST :(")
		break
	case client.ONGOING:
		log.Println("WTF !!")
		break
	}
}
